sklearn/ensemble/tests/test_stacking_classifier.py

```diff
@@ -0,0 +1,217 @@
+    assert_array_equal(eclf1.predict_proba(X), eclf2.predict_proba(X))
+
+
+def test_sample_weight():
+    """Tests sample_weight parameter of StackingClassifier"""
+    decimal = 10
+
+    clf1 = LogisticRegression(random_state=123)
+    clf2 = RandomForestClassifier(random_state=123)
+    clf3 = GaussianNB()
+    clfm = GaussianNB()
+    eclf1 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm
+    ).fit(X, y, sample_weight=np.ones((len(y),)))
+    eclf2 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm
+    ).fit(X, y)
+    assert_array_almost_equal(eclf1.predict(X), eclf2.predict(X),
+                              decimal=decimal)
+    assert_array_almost_equal(eclf1.predict_proba(X), eclf2.predict_proba(X),
+                              decimal=decimal)
+
+    sample_weight = np.random.RandomState(123).uniform(size=(len(y),))
+    clf4 = KNeighborsClassifier()
+    eclf3 = StackingClassifier(
+        estimators=[('lr', clf1), ('knn', clf4)],
+        meta_estimator=clfm)
+    msg = ("Underlying estimator knn of type "
+           "<class 'sklearn.neighbors.classification.KNeighborsClassifier'> "
+           "does not support `sample_weight`.")
+    assert_raise_message(ValueError, msg, eclf3.fit, X, y,
+                         sample_weight=sample_weight)
+    eclf4 = StackingClassifier(
+        estimators=[('lr', clf1), ('nb', clf3)],
+        meta_estimator=clf4)
+    msg = ("Underlying meta estimator "
+           "<class 'sklearn.neighbors.classification.KNeighborsClassifier'> "
+           "does not support `sample_weight`.")
+    assert_raise_message(ValueError, msg, eclf4.fit, X, y,
+                         sample_weight=sample_weight)
+
+
+def test_classify_iris():
+    """Check classification by majority label on dataset iris."""
+    clf1 = LogisticRegression(random_state=123)
+    clf2 = RandomForestClassifier(random_state=123)
+    clf3 = GaussianNB()
+    clfm = GaussianNB()
+    eclf = StackingClassifier(estimators=[('lg', clf1),
+                                          ('rf', clf2),
+                                          ('nb', clf3)],
+                              meta_estimator=clfm)
+    scores = cross_val_score(eclf, X, y, cv=5, scoring='accuracy')
+    assert_almost_equal(scores.mean(), 0.95, decimal=2)
+
+
+def test_predict_on_toy_problem():
+    """Manually check predicted class labels for toy dataset."""
+    clf1 = LogisticRegression(random_state=123)
+    clf2 = RandomForestClassifier(random_state=123)
+    clf3 = GaussianNB()
+
+    X = np.array([[-1.1, -1.5],
+                  [-1.2, -1.4],
+                  [-3.4, -2.2],
+                  [1.1, 1.2],
+                  [2.1, 1.4],
+                  [3.1, 2.3]])
+
+    y = np.array([1, 1, 1, 2, 2, 2])
+
+    assert_equal(all(clf1.fit(X, y).predict(X)), all([1, 1, 1, 2, 2, 2]))
```

Comment:
do you mean `assert_array_equal`? `all` will return a boolean, so you're asserting the equality of booleans here.